### PR TITLE
`impl Data` for a bunch of std types.

### PR DESCRIPTION
**Pull request by derekdreery**
This patch implements `Data` for some more types in `std`. My motivation was `SystemTime`, but I thought I may as well implement it anywhere else it made sense.

### CODE REVIEW

File: druid/src/data.rs [hunk 1]
```diff
@@ -269,6 +294,89 @@ impl<T0: Data, T1: Data, T2: Data, T3: Data, T4: Data, T5: Data> Data for (T0, T
...
+        ptr::eq(*self, *other)
+    }
+}
+
+impl<T: 'static + ?Sized> Data for *mut T {
+    fn same(&self, other: &Self) -> bool {
+        // This Does not dereference the pointer, only the reference to the pointer.
+        ptr::eq(*self, *other)
+    }
+}
```
Comment (Maan2003) [inline druid/src/data.rs:309]: I would leave these ones. When someone has raw pointers, they are better using `same_fn`. But if you add these, then also add `NonNull`

**Review by Maan2003 — COMMENTED** [review]
Looks good to me. But I am not sure about the raw pointers.

File: druid/src/data.rs (hunk 1 above)
Comment (derekdreery) [reply druid/src/data.rs:309]: Hmm OK I'm sold on removing the raw pointers.

Comment (cmyr) [reply druid/src/data.rs]: yes I agree I struggle to think of a good case where you'd want this.

**Review by cmyr — APPROVED** [review]
This makes sense, thanks!

**Comment by derekdreery** [comment]
I had a look at implementing it for `Pin`, but the types were complex, and I'm not sure exactly what it means to ptr-compare pinned things. Can always be added later anyway.

**Pull request merged**